.github/workflows/docs.yml

```diff
@@ -39,34 +42,51 @@ jobs:
       - name: Create Virtual Environment
         run: hatch env create docs
 
-      - name: "Check for mkdocs build --strict"
-        # XXX Enable strict mode once docs are clean
-        run: |
-          hatch run docs:build
-          # hatch run docs:build --strict
-        # If pull request or not master branch and not a tag
-        if: github.event_name == 'pull_request' || (github.event_name == 'push' && github.ref != 'refs/heads/master' && !startsWith(github.ref, 'refs/tags/')) || github.event_name == 'workflow_dispatch'
-
       - name: Configure git
         run: |
           git config user.name github-actions
           git config user.email github-actions@github.com
 
-      - name: Deploy docs (dev)
-        if: github.event_name == 'push' && github.ref_name == 'master' && github.ref_type == 'branch'
+      - name: Print variables for debugging
+        # XXX this step may be removed
+        # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context
+        run: |
+          # env vars used in this action
+          echo 'event_name  : ' ${{ github.event_name }}
+          echo 'event.action: ' ${{ github.event.action }}
+          echo 'ref         : ' ${{ github.ref }}
+          echo 'ref_name    : ' ${{ github.ref_name }}
+          echo 'ref_type    : ' ${{ github.ref_type }}
+          echo 'repository  : ' ${{ github.repository }}
+
+      # Conditions make sure to select the right step, depending on the job trigger.
+      # Only one of the steps below will run. The others will be skipped.
+
+      - name: Check docs in pull requests with strict mode
+        if: github.event_name == 'pull_request'
+        run: |
+          # XXX Enable strict mode once docs are clean
+          echo "Strict check of docs disabled."
+          # hatch run docs:build --strict
+
+      - name: Build & deploy "dev" docs for a new commit to master
+        if: github.event_name == 'push' && github.ref_type != 'tag'
         run: |
           export SHORT_SHA=$(echo "${GITHUB_SHA}" | cut -c1-7)
-          hatch run docs:mike deploy dev --update-aliases --title "dev (${SHORT_SHA})" --push --alias-type=redirect
+          hatch run docs:mike deploy --push --update-aliases --title "dev (${SHORT_SHA})" dev
 
-      - name: Deploy docs (tag)
+      - name: Build & deploy docs for a new tag
         if: github.ref_type == 'tag' && github.event_name == 'push'
         run: |
-          hatch run docs:mike deploy ${{ github.ref_name }} latest --push --update-aliases --alias-type=redirect
+          # XXX next line removes docs wrongly deployed under latest identifier; remove once latest is used only as alias
```

Comment:
What is meant by the `"docs wrongly deployed under latest identifier; remove once latest is used only as alias"`? And how do we know when the `latest` identifier is removed?

Reply:
mike deploys versions under an identifier and optionally also gives an alias to the identifier. Identifiers cannot be updated but have to be deleted and then recreated. Aliases on the other hand can be updated which means a new alias can be given to an existing identifier. For whatever reason, in current gh-pages "latest" was created as an identifier (there is a directory latest) which is not what we want; instead the alias latest should be attached to a version identifier.